Detectors/MUON/MID/Raw/src/Encoder.cxx

```diff
@@ -157,9 +156,11 @@ void Encoder::process(gsl::span<const ColumnData> data, InteractionRecord ir, Ev
   /// Encodes data
 
   // The CTP trigger arrives to the electronics with a delay
-  applyElectronicsDelay(ir.orbit, ir.bc, -mElectronicsDelay.localToBC);
+  if (ir.differenceInBC(mRawWriter.getHBFUtils().getFirstSampledTFIR()) > mElectronicsDelay.localToBC) { // RS: not sure this is correct.
+    applyElectronicsDelay(ir.orbit, ir.bc, -mElectronicsDelay.localToBC);
+  }
```

Comment:
@dstocco : You are subtracting from the `ir` a few 10s BCs, as a result, it can become negative (i.e. wrap to a huge value) or precede the 1st sampled TF. Could you check this part?

The example where this was happening (encountered by @davidrohr ) had the following:
1st sampled orbit: 256
Sampled collisions:
```
root [3] for (auto ir : irc) std::cout << ir << "\n"
BCid:   48 Orbit:    256 |T in BC(ns): -0.069
BCid: 1346 Orbit:    256 |T in BC(ns): 0.040
BCid: 1998 Orbit:    256 |T in BC(ns): -0.206
BCid: 1356 Orbit:    257 |T in BC(ns): -0.028
BCid:   42 Orbit:    258 |T in BC(ns): 0.129
```
Produced MID digits ROFs:
```
root [1] o2sim->Scan("MIDROFRecords.interactionRecord.bc : MIDROFRecords.interactionRecord.orbit")
***********************************************
*    Row   * Instance * MIDROFRec *  MIDROFRe *
***********************************************
*        0 *        0 *        48 *       256 *
*        0 *        1 *      1346 *       256 *
*        0 *        2 *      1356 *       257 *
*        0 *        3 *        42 *       258 *
***********************************************
```
Subtracting from the 1st ROF IR `mElectronicsDelay.localToBC = 59` you call 1st addData for the orbit 255  which appears in the TF preceding the generated one.